src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
       deep_copy(ret)
     end
 
+
+    private
+
+    def ReallyAbort
+      Popup.ReallyAbort(true)
+    end
+
+
+    # Read settings dialog
+    # @return `abort if aborted and `next otherwise
+    def ReadDialog
+      return :abort if !Confirm.MustBeRoot
+
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "read", ""))
+      ret = Snapper.Init()
+      ret ? :next : :abort
+    end
+
+    def open_modify_dialog(content)
+      UI.OpenDialog(
+        Opt(:decorated),
+        HBox(
+          HSpacing(1),
+          VBox(
+            VSpacing(0.5),
+            HSpacing(65),
+            content,
+            VSpacing(0.5),
+            ButtonBox(
+              PushButton(Id(:ok), Label.OKButton),
+              PushButton(Id(:cancel), Label.CancelButton)
+            ),
+            VSpacing(0.5)
+          ),
+          HSpacing(1)
+        )
+      )
+    end
+
+    def snapshot_term(prefix, data)
+      data = deep_copy(data)
+      HBox(
+        HSpacing(),
+        Frame(
+          "",
+          HBox(
+            HSpacing(0.4),
+            VBox(
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "description")),
```

Comment:
for new code I think it is better to use simple `prefix + "description"` instead of `Ops.add`